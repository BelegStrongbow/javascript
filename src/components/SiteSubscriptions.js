import React from "react";
import styled from "styled-components";
import SeoIcon from "../icons/Yoast/Yoast_SEO_Icon_Small.svg";
import LocalIcon from "../icons/Yoast/Local_SEO_Icon_Small.svg";
import NewsIcon from "../icons/Yoast/News_Icon_Small.svg";
import VideoIcon from "../icons/Yoast/Video_SEO_Icon_Small.svg";
import WooIcon from "../icons/Yoast/Woo_Icon_Small.svg";
import LocalWooIcon from "../icons/Yoast/Local_Woo_Icon_Small.svg";

let YoastProducts = {
	seo: { name: "Yoast SEO", image: SeoIcon },
	local: { name: "Local SEO", image: LocalIcon },
	video: { name: "Video SEO", image: VideoIcon },
	news: { name: "News SEO", image: NewsIcon },
	woo: { name: "WooCommerce SEO", image: WooIcon },
	localwoo: { name: "Local SEO for Woo", image: LocalWooIcon },
};

const SiteSubscriptionIcons = styled.span`
	background-image: url(${ props => props.image });
	opacity: ${ props => props.isActive ? 1.0 : 0.2 };
	width: 40px;
	height: 40px;
	float: left;
	margin: 0 5px;
`;

SiteSubscriptionIcons.propTypes = {
	image: React.PropTypes.string.isRequired,
	isActive: React.PropTypes.bool.isRequired,
};

/**
 * Renders a subscriptions component.
 *
 * @param {Object} props The props to use.
 * @returns {XML} The rendered Subscriptions component.
 */
export default function SiteSubscriptions( props ) {
	return (
		<div>
			{
				Object.keys( YoastProducts ).map( function( productName ) {
					let isActive = props.activeSubscriptions.includes( productName );
					let product = YoastProducts[ productName ];

					return (
						<SiteSubscriptionIcons
							key={ productName }
							image={ product.image }
							isActive={ isActive }
							aria-label={ isActive ? product.name + " is active" : product.name + " is inactive" }
							role="img"
						/>
					);
				} )
			}
		</div>
	);
}

SiteSubscriptions.propTypes = {
	activeSubscriptions: React.PropTypes.arrayOf( React.PropTypes.string ),
};

SiteSubscriptions.defaultProps = {
	activeSubscriptions: [],
};
