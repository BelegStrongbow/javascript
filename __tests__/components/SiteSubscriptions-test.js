import React from 'react';
import renderer from 'react-test-renderer';
import { createComponentWithIntl } from "../../utils";

import SiteSubscriptions from '../../src/components/SiteSubscriptions';

let plugins = [
	{
		glNumber: 111,
		ids: [ "1" ],
		icon: "test.jpg",
		name: "Test",
	},
	{
		glNumber: 222,
		ids: [ "2" ],
		icon: "test.jpg",
		name: "Test2",
	},
	{
		glNumber: 333,
		ids: [ "3" ],
		icon: "test.jpg",
		name: "Test3",
	},
	{
		glNumber: 444,
		ids: [ "4" ],
		icon: "test.jpg",
		name: "Test4",
	}
];

let activeSubscriptions = [
	{
		productId: "1",
	},
	{
		productId: "2",
	},
];

test('the subscription component matches the snapshot', () => {
	const component = createComponentWithIntl(
		<SiteSubscriptions plugins={ plugins } activeSubscriptions={ activeSubscriptions } />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('the subscription component without plugins with active subscriptions using the default matches the snapshot', () => {
	const component = createComponentWithIntl(
		<SiteSubscriptions activeSubscriptions={ activeSubscriptions } />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('the subscription component with plugins without active subscriptions using the default matches the snapshot', () => {
	const component = createComponentWithIntl(
		<SiteSubscriptions plugins={ plugins } />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('the subscription component without plugins or active subscriptions using the default matches the snapshot', () => {
	const component = createComponentWithIntl(
		<SiteSubscriptions />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
