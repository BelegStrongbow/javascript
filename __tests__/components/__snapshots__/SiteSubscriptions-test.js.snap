exports[`test the subscription component matches the snapshot 1`] = `
<span>
  <span
    aria-label="Yoast SEO is inactive"
    className="hGymDt"
    role="img" />
  <span
    aria-label="Local SEO is inactive"
    className="kswMOV"
    role="img" />
  <span
    aria-label="Video SEO is active"
    className="jzGgcX"
    role="img" />
  <span
    aria-label="News SEO is inactive"
    className="kdRmgg"
    role="img" />
  <span
    aria-label="WooCommerce SEO is active"
    className="gKxAXd"
    role="img" />
  <span
    aria-label="Local SEO for Woo is inactive"
    className="egmryf"
    role="img" />
</span>
`;

exports[`test the subscription component without active subscriptions using the default matches the snapshot 1`] = `
<span>
  <span
    aria-label="Yoast SEO is inactive"
    className="hGymDt"
    role="img" />
  <span
    aria-label="Local SEO is inactive"
    className="kswMOV"
    role="img" />
  <span
    aria-label="Video SEO is inactive"
    className="icgIoi"
    role="img" />
  <span
    aria-label="News SEO is inactive"
    className="kdRmgg"
    role="img" />
  <span
    aria-label="WooCommerce SEO is inactive"
    className="dritTx"
    role="img" />
  <span
    aria-label="Local SEO for Woo is inactive"
    className="egmryf"
    role="img" />
</span>
`;
